README.md

```diff
@@ -1,6 +1,9 @@
 ml-boilerplate
 ==============================
 
+[![PyLint](https://github.com/gianfrancodemarco/ml-boilerplate/actions/workflows/code_checks.yml/badge.svg?branch=main)](https://github.com/Francesco-Ranieri/ml-boilerplate/actions/workflows/code_checks.yml)
+[![PynbLint](https://github.com/gianfrancodemarco/ml-boilerplate/actions/workflows/code_checks.yml/badge.svg?branch=main)](https://github.com/Francesco-Ranieri/ml-boilerplate/actions/workflows/code_checks.yml)
+
```

Comment:
You are only pushing the link, not the code to generate the badge.
Also the link is wrong.

Reply:
For standard Action badge, you can copy the md code in the "Actions" section and paste it in the readme.md file.
<br>
PS:
The push trigger also intercepts the merge action because a merged PR will always create a push event to the branch it's being merged onto. [[source]](https://stackoverflow.com/questions/60710209/trigger-github-actions-only-when-pr-is-merged)